src/autopredictor/bestscore.py

```diff
@@ -13,6 +16,44 @@ def display_best_score(X, scoring_metric):
     Returns
     -------
     DataFrame
-        A DataFrame containing the best score with respect to a specific scoring metric alongside the corresponding 
-        model.
-    """
\ No newline at end of file
+        If the scoring metric is found, a dataframe containing the best score and the corresponding model is returned.
+        If the scoring metric is not found, a ValueError is raised.
+
+    Examples
+    --------
+    >>> from autopredictor.bestscore import display_best_score
+    >>> df = pd.DataFrame({'MAE': [5.6, 3.4],
+                                  'MSE': [9.4, 21.4],
+                                  'MAPE': [0.34, 0.45]},
+                                 index=['Linear Regression', 'Random Forest'])
+    >>> display_best_score(df, 'MAE')
+                       MAE  
+    Linear Regression  5.6
+    
+    >>> display_best_score(df, 'R2')
+    ValueError: Invalid Scoring metric 'R2'.The specified metric is not in the list of available metrics. Available metrics: MAE, MSE, MAPE.
+   """
+    if X is None or not isinstance(X, pd.DataFrame):
+        raise TypeError("Invalid DataFrame provided.")
+    
+    if X.empty:
+        raise TypeError("DataFrame is empty.")
+    
+    if not isinstance(scoring_metric, str):
+        raise ValueError("scoring_metric must be a string.")
+
+    if scoring_metric not in X.columns:
+        available_metrics = X.columns.tolist()
+        available_metrics_string = ", ".join(available_metrics)
+        raise ValueError (f"Invalid Scoring metric '{scoring_metric}'.The specified metric is not in the list of available metrics. Available metrics: {available_metrics_string}.")
```

Comment:
```suggestion
        raise ValueError (f"Invalid Scoring metric '{scoring_metric}'. The specified metric is not in the list of available metrics. Available metrics: {available_metrics_string}.")
```
Grammar (if changed here, it must be changed in the test file)